crates/polars-core/src/chunked_array/logical/categorical/builder.rs

```diff
@@ -96,10 +97,15 @@ impl RevMapping {
     pub fn get_categories(&self) -> &Utf8Array<i64> {
         match self {
             Self::Global(_, a, _) => a,
-            Self::Local(a) => a,
+            Self::Local(a, _) => a,
         }
     }
 
+    pub fn build_local(categories: Utf8Array<i64>) -> RevMapping {
+        let hash = RandomState::with_seed(0).hash_one(categories.values().as_slice());
```

Comment:
We must also hash the indexes here as now `["ab", "c]` collides with `["a", "b", "c"]`. then we can concat the two u64 into a u128 to reduce some collision probablity.